app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class ContactsController < ApplicationController
+  before_action :require_login
+  def index
+    @company =found_company
+  end
+
+  def show
+
+    @company =found_company
+    @contact = @company.contacts.find(params[:id])
+
+    render :show
+  end
+
+  def new
+    @company = found_company
+    @contact = Contact.new
+  end
+
+  def create
+    @company = found_company
+    if @company.contacts.create(contact_params)
+      redirect_to company_contacts_path(@company)
+    else
+      render :new
```

Comment:
Don't forget to handle the errors. What happens if a user enters something incorrectly and the Contact doesn't get created? The user should see a nice error message.